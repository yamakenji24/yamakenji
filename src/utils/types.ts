export interface ExperienceType {
  id: string;
  during: string;
  title: string;
  body: string;
}

export interface SkillType {
  id: string;
  name: string;
  img: string;
}

export interface OGPDataType {
  url: string;
  title: string;
  image: OpenGraphImage;
}

interface OpenGraphImage {
  height: string;
  type: string;
  url: string;
  width: string;
}

export interface WorkType {
  id: string;
  github: string;
  link: string;
  img: string;
  title: string;
  skills: Array<{
    fieldId: string;
    skill: string;
  }>;
  body: string;
}

export interface GetWorksResponse {
  contents: Array<{
    id: string;
    github: string;
    link: string;
    title: string;
    skills: Array<{
      fieldId: string;
      skill: string;
    }>;
    body: string;
    img: {
      url: string;
      height: number;
      width: number;
    };
    createdAt: string;
    updatedAt: string;
    publishedAt: string;
    revisedAt: string;
  }>;
  totalCount: number;
  offset: number;
  limit: number;
}

export interface GetArticlesResponse {
  contents: Array<{
    id: string;
    url: string;
    title: string;
    img: {
      url: string;
      height: number;
      width: number;
    };
    date: string;
    createdAt: string;
    updatedAt: string;
    publishedAt: string;
    revisedAt: string;
  }>;
  totalCount: number;
  offset: number;
  limit: number;
}