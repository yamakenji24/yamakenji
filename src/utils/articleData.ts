export interface ArticleData {
  url: string;
  title: string;
  img: string;
  date: string;
}
export const articleData: Array<ArticleData> = [
  {
    url: 'https://zenn.dev/yamakenji24/articles/2921714d4b69d0',
    title: 'Next.jsとTailwindを利用したエンジニア名刺アプリ',
    img: '/article/cardcreator.jpg',
    date: '2020/12',
  },
  {
    url: 'https://yamakenji24.hatenablog.com/entry/2020/11/13/192543',
    title: 'Chatworkのサマーインターン2020に参加してきた',
    img: '/article/intern.jpg',
    date: '2020/09',
  },
  {
    url: 'https://shell-mag.com/vol-68/',
    title: 'シェルスクリプトマガジン Vol.68',
    img: '/article/shellvol68.jpg',
    date: '2020/08',
  },
  {
    url: 'https://qiita.com/yamakenji/items/38fd1175bdd3338fb810',
    title: 'GraphQLについてまとめてみた',
    img: '/article/GraphQL.jpg',
    date: '2020/06/23',
  },
  {
    url: 'https://qiita.com/yamakenji/items/b94fa2ab104679b29b3a',
    title: 'React+Reduxで音楽検索アプリを作ってみた',
    img: '/article/spotify.jpg',
    date: '2019/12/14',
  },
  {
    url: 'https://shell-mag.com/vol-60/',
    title: 'シェルスクリプトマガジン Vol.60',
    img: '/article/shellvol60.jpg',
    date: '2019/06',
  },
];
