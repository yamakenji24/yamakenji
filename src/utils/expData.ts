export interface ExpData {
  during: string;
  title: string;
  body: string;
}

export const expData: Array<ExpData> = [
  {
    during: 'September 14th, 2020 ~ September 18th, 2020',
    title: 'Monotaro Intern',
    body: 'インターンに参加 ~',
  },
  {
    during: 'August 17th, 2020 ~ September 4th, 2020',
    title: 'ChatWork Intern',
    body: 'インターンに参加 ~',
  },
  {
    during: 'April 2020 - present',
    title: 'Kagawa University',
    body: 'Master: Division of Reliability-based Information Systems Engineering',
  },
  {
    during: 'April 2016 ~ March 2020',
    title: 'Kagawa University',
    body: 'Bachelor: Faculty of Engineering',
  },
];
