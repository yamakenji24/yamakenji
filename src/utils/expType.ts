export interface ExperienceType {
  id: string;
  during: string;
  title: string;
  body: string;
}

export interface SkillType {
  id: string;
  name: string;
  img: string;
}
