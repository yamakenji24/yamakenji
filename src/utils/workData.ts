export interface WorkType {
  id: string;
  github: string;
  link: string;
  img: string;
  title: string;
  skills: Array<{
    fieldId: string;
    skill: string;
  }>;
  body: string;
}

export interface GetWorksResponse {
  contents: Array<{
    id: string;
    github: string;
    link: string;
    title: string;
    skills: Array<{
      fieldId: string;
      skill: string;
    }>;
    body: string;
    img: {
      url: string;
      height: number;
      width: number;
    };
    createdAt: string;
    updatedAt: string;
    publishedAt: string;
    revisedAt: string;
  }>;
  totalCount: number;
  offset: number;
  limit: number;
}