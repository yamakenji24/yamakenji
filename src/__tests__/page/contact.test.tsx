import React from 'react';
import { render } from '../utils';

import Contact from 'pages/contact';

test('/contact', () => {
  const { container } = render(<Contact />);
  expect(container).toMatchSnapshot();
});
