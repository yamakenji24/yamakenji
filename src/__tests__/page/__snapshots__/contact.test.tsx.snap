// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/contact 1`] = `
<div>
  <div
    class="css-0"
  >
    <nav
      class="css-n09xmb"
    >
      <div
        class="css-1lipxjv"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjAiIGhlaWdodD0iNTAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <img
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </div>
      </div>
      <div
        class="css-rfbghr"
      >
        <p
          class="chakra-text css-y8gvm7"
        >
          <a
            class="chakra-link css-3wpten"
            data-test-id="navigation-Top"
          >
            Top
          </a>
        </p>
        <p
          class="chakra-text css-y8gvm7"
        >
          <a
            class="chakra-link css-3wpten"
            data-test-id="navigation-Work"
          >
            Work
          </a>
        </p>
        <p
          class="chakra-text css-y8gvm7"
        >
          <a
            class="chakra-link css-3wpten"
            data-test-id="navigation-Article"
          >
            Article
          </a>
        </p>
        <p
          class="chakra-text css-y8gvm7"
        >
          <a
            class="chakra-link css-3wpten"
            data-test-id="navigation-Contact"
          >
            Contact
          </a>
        </p>
        <a
          class="chakra-link css-f4h6uy"
        >
          <svg
            class="chakra-icon css-tvkuyh"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
            />
          </svg>
        </a>
      </div>
    </nav>
    <main>
      <div
        class="css-0"
      >
        <h4
          class="chakra-heading css-1sr6c4u"
        >
          Contact
        </h4>
        <div
          class="chakra-container css-ypmdfm"
        >
          <p
            class="chakra-text css-xi606m"
          >
            何かございましたら下のフォームからご連絡ください。
            <br />
            Feel free to contact me from below.
          </p>
          <div
            class="css-1dfptg7"
          >
            <div
              class="chakra-form-control css-1kxonj9"
              role="group"
            >
              <input
                aria-required="true"
                class="chakra-input css-1lw1oo1"
                id="name"
                isfullwidth="true"
                placeholder="Name"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="chakra-form-control css-1kxonj9"
              role="group"
            >
              <input
                aria-required="true"
                class="chakra-input css-1lw1oo1"
                id="title"
                isfullwidth="true"
                placeholder="Title"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="chakra-form-control css-1kxonj9"
              role="group"
            >
              <input
                aria-required="true"
                class="chakra-input css-1lw1oo1"
                id="email"
                isfullwidth="true"
                placeholder="Email Address"
                required=""
                type="email"
                value=""
              />
            </div>
            <div
              class="chakra-form-control css-1kxonj9"
              role="group"
            >
              <input
                aria-required="true"
                class="chakra-input css-1lw1oo1"
                id="message"
                isfullwidth="true"
                multiline="true"
                placeholder="Contents"
                required=""
                value=""
              />
            </div>
            <button
              class="chakra-button css-1oaq8qa"
              type="button"
            >
              送信
            </button>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
